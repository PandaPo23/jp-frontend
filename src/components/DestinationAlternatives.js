import React from 'react';

import { Alternatives, FlexPanel } from '../design-system';

export const DestinationAlternatives = ({ alternatives }) => (
  <FlexPanel
    bg="background"
    scrollable
    p={4}
    width={1}
    flex={1}
    flexWrap="wrap"
    flexFlow="row wrap"
    flexDirection="row"
    justifyContent="space-between"
  >
    {alternatives.map((al, key) => (
      <Alternatives key={key} name={al.name} imageUrl={al.image} />
    ))}
  </FlexPanel>
);

export default DestinationAlternatives;
