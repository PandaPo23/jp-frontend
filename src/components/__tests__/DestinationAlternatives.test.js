import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { DestinationAlternatives } from '../DestinationAlternatives';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({}),
}));
const alternatives = [
  {
    id: 'sydney',
    name: 'Sydney',
    image: 'https://source.unsplash.com//800x600/?sydney',
  },
  {
    id: 'melbourne',
    name: 'Melbourne',
    image: 'https://source.unsplash.com/1600x900/?melbourne',
  },
  {
    id: 'adelaide',
    name: 'Adelaide',
    image: 'https://source.unsplash.com/1600x900/?adelaide',
  },
  {
    id: 'newcastle',
    name: 'Newcastle',
    image: 'https://source.unsplash.com/1600x900/?newcastle',
  },
  {
    id: 'gold_coast',
    name: 'Gold Coast',
    image: 'https://source.unsplash.com/1600x900/?gold-coast',
  },
  {
    id: 'canberra',
    name: 'Canberra',
    image: 'https://source.unsplash.com/1600x900/?canberra',
  },
];

describe('DestinationAlternatives component', () => {
  it('renders DestinationAlternatives correctly', () => {
    const localProps = {
      alternatives,
    };
    const wrapper = shallow(<DestinationAlternatives {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
