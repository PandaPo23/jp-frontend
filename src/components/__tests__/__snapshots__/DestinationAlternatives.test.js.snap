// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationAlternatives component renders DestinationAlternatives correctly 1`] = `
<FlexPanel
  bg="background"
  display="flex"
  flex={1}
  flexDirection="row"
  flexFlow="row wrap"
  flexWrap="wrap"
  justifyContent="space-between"
  p={4}
  scrollable={true}
  width={1}
>
  <Alternatives
    imageUrl="https://source.unsplash.com//800x600/?sydney"
    key="0"
    name="Sydney"
  />
  <Alternatives
    imageUrl="https://source.unsplash.com/1600x900/?melbourne"
    key="1"
    name="Melbourne"
  />
  <Alternatives
    imageUrl="https://source.unsplash.com/1600x900/?adelaide"
    key="2"
    name="Adelaide"
  />
  <Alternatives
    imageUrl="https://source.unsplash.com/1600x900/?newcastle"
    key="3"
    name="Newcastle"
  />
  <Alternatives
    imageUrl="https://source.unsplash.com/1600x900/?gold-coast"
    key="4"
    name="Gold Coast"
  />
  <Alternatives
    imageUrl="https://source.unsplash.com/1600x900/?canberra"
    key="5"
    name="Canberra"
  />
</FlexPanel>
`;
